src/Controller/CalendarController.php

```diff
@@ -0,0 +1,413 @@
+        $j->debut=$debutSQL;
+        $j->fin=$finSQL;
+        $j->index= "date";
+        $j->fetch();
+        $joursFeries=$j->elements;
+
+        //Sélection des horaires de travail
+        $db = new \db();
+        $db->select2("personnel","temps", array("id"=> $perso_id));
+        $temps=json_decode(html_entity_decode($db->result[0]['temps'], ENT_QUOTES|ENT_IGNORE, 'UTF-8'), true); //$temps = emploi du temps
+
+        // Sélection des absences
+        $filter = $this->config('Absences-validation')?"AND `valide`>0":null;
+        $db = new \db();
+        $db->select("absences",null, "`perso_id`='$perso_id' $filter");
+        $absences =$db->result; //absences = tableau d'absences
+
+        //Plannings verrouillés
+        $verrou = array();
+        $nbSites = $this->config('Multisites-nombre');
+        for ($i = 1; $i <= $nbSites; $i++){
+            $verrou[$i]=array();
+        }
+
+        $db = new \db();
+        $db->select2("absences", null, "`perso_id`=$perso_id' $filter");
+        if ($db->result){
+            foreach ($db->result as $elem){
+                $verrou[$elem('site')][]=$elem['date'];
+            }
+        }
+
+        //Sélection des postes occupés
+        $db= new \db();
+        $db->selectInnerJoin(
+            array("pl_poste", "poste"),
+            array("postes", "id"),
+            array("date","debut","fin","absent","site"),
+            array(array("name"=>"nom", "as"=>"poste")),
+            array("perso_id"=>$perso_id, "date"=>"BETWEEN $debutSQL AND $finSQL"),
+            array(),
+            "ORDER BY date, debut, fin, site, poste"
+        );
+        $postes = $db->result;
+
+        // Affiche des cellules vides devant le premier jour demandé de façon à avoir les lundis dans la première colonne
+        $d = new \datePl($debutSQL);
+        $cellsBefore = $d->position>0?$d->position-1:6;
+
+        $nb=$cellsBefore;
+        $current=$debutSQL;
+        $days = array();
+        while ($current<=$finSQL) {
+            $current_postes=array();
+            $date_tab=explode("-", $current);
+            $date_aff=dateAlpha($current, false, false);
+            $jour=date("w", strtotime($current))-1;
+            $d=new \datePl($current);
+            $semaine=$d->semaine;
+            $j1=$d->dates[0];
+
+            if ($jour<0) {
+                $jour=6;
+            }
+
+            // Si utilisation de 2 ou 3 plannings hebdo, hors EDTSamedi
+            if (!$this->config('EDTSamedi')) {
+                if ($d->semaine3==2) {
+                    $jour=$jour+7;
+                } elseif ($d->semaine3==3) {
+                    $jour=$jour+14;
+                }
+            }
+
+            // Si utilisation d'un planning pour les semaines sans samedi et un planning pour les semaines avec samedi travaillé
+            if ($this->config('EDTSamedi')) {
+              // Pour chaque agent, recherche si la semaine courante est avec samedi travaillé ou non
```

Comment:
Indentation

Reply:
Which element is concerned in this extract ?